brownie/network/contract.py

```diff
@@ -127,6 +128,66 @@ def decode_input(self, calldata: Union[str, bytes]) -> Tuple[str, Any]:
 
         return function_sig, input_args
 
+    def flatten(self) -> dict:
+        """
+        Return a dict with a single, deployable source code for this contract
+        and further information needed for verification
+        """
+        language = self._build["language"]
+        if language == "Vyper":
+            return self._build["source"]
+        elif language == "Solidity":
+            flattened_source = ""
+
+            for name in self._build["dependencies"]:
+                build_json = self._project._build.get(name)
+                offset = slice(*build_json["offset"])
+                source = self._slice_source(build_json["source"], offset)
+                file_name = Path(build_json["sourcePath"]).parts[-1]
+                flattened_source = f"{flattened_source}\n\n// File: {file_name}\n\n{source}"
+
+            build_json = self._build
+            version = build_json["compiler"]["version"]
+            version_short = re.findall(r"^[^+]+", version)[0]
+            offset = slice(*build_json["offset"])
+            source = self._slice_source(build_json["source"], offset)
+            file_name = Path(build_json["sourcePath"]).parts[-1]
+            licenses = re.findall(
+                r"SPDX-License-Identifier:(.*)\n", build_json["source"][: offset.start]
+            )
+            license_identifier = licenses[0].strip() if len(licenses) == 1 else "NONE"
+            flattened_source = (
+                f"// SPDX-License-Identifier: {license_identifier}\n\n"
+                f"pragma solidity {version_short};{flattened_source}\n\n"
+                f"// File: {file_name}\n\n{source}\n"
+            )
+            return {
```

Comment:
I understand the importance of returning all this information for source verification, but with Vyper the function is only returning a source file.  In order to keep the return value consistent it might be better to simply raise when language == Vyper and say "you don't need to flatten this".  ?

Reply:
Agreed, as long as vyper verification is not implemented via API, we should raise.